ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -336,6 +337,10 @@ public void onCompletion(Long result, Exception exception) {
         if (exception != null) {
           setOperationException(exception);
         }
+        ResponseInfo responseInfo = chunkIndexToResponseInfo.remove(numChunksWrittenOut);
+        if (responseInfo != null) {
+          responseInfo.release();
+        }
         numChunksWrittenOut++;
```

Comment:
optional: `numChunksWrittenOut` can be atomic.

Reply:
updated.